### ISSUE TITLE
Style/ArgumentsForwarding breaks code when assigning to an array

### ISSUE THREAD

**Issue by Adrian-Hirt** [rubocop/rubocop#12104]
When running `Style/ArgumentsForwarding` on some code that assigns the result of a method call to an array element, the autocorrect function creates broken code.

Consider the following example:

```ruby
class Foo
  def self.bar(*args)
    result = []
    result[0] = new(*args)

    result
  end
end
```

--------

## Expected behavior

Rubocop should change the code as follows:

```ruby
class Foo
  def self.bar(*)
    result = []
    result[0] = new(*)

    result
  end
end
```

## Actual behavior

The following code is generated:

```ruby
class Foo
  def self.bar(*)
    result = []
    result[0](= new(*))

    result
  end
end
```

```
Offenses:

foo.rb:2:16: C: [Corrected] Style/ArgumentsForwarding: Use anonymous positional arguments forwarding (*).
  def self.bar(*args)
               ^^^^^
foo.rb:4:14: F: Lint/Syntax: unexpected token tLPAREN2
(Using Ruby 3.2 parser; configure using TargetRubyVersion parameter, under AllCops)
    result[0](= new(*))
             ^
foo.rb:4:21: C: [Corrected] Style/ArgumentsForwarding: Use anonymous positional arguments forwarding (*).
    result[0] = new(*args)
                    ^^^^^
foo.rb:4:23: F: Lint/Syntax: unexpected token tRPAREN
(Using Ruby 3.2 parser; configure using TargetRubyVersion parameter, under AllCops)
    result[0](= new(*))
                      ^
```

## Steps to reproduce the problem

1) Run rubocop on the above example file

## RuboCop version

```
$ [bundle exec] rubocop -V
1.55.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.1) [x86_64-linux]
  - rubocop-rails 2.15.2
```

## Additional information

I assume the error stems from https://github.com/rubocop/rubocop/blob/f3790afce259aec75bb94d571803c7e5488f1807/lib/rubocop/cop/style/arguments_forwarding.rb#L189 
Assigning to an array element is treated the same as if a method was called, which makes rubocop add the parentheses.

Running the autocorrect on 

```ruby
class Foo
  def self.bar(*args)
    baz new(*args)
  end
end
```

generates the correct code

```ruby
class Foo
  def self.bar(*)
    baz(new(*))
  end
end
```

However for assigning to an array, the parentheses should not be added.

Please let me know if you need any additional informations to be able to debug this issue.

**Comment by koic** [comment]
This issue was fixed by RuboCop 1.55.1. Can you upgrade to the latest version? Thank you.

**Comment by Adrian-Hirt** [comment]
Thank you for the fast reply, it indeed works with `1.55.1`. 👍
